fs/accounting.go

```diff
@@ -394,12 +444,12 @@ func (acc *Account) read(in io.Reader, p []byte) (n int, err error) {
 	// Get the token bucket in use
 	tokenBucketMu.Lock()
 	tb := tokenBucket
-	tokenBucketMu.Unlock()
 
 	// Limit the transfer speed if required
 	if tb != nil {
 		tb.Wait(int64(n))
 	}
+	tokenBucketMu.Unlock()
```

Comment:
Ohhh, this has a story! :) I thought the same and initially it would lock tokenBucket, grab a copy, unlock and then Wait. This caused the *entire* program to sit idle, doing nothing, after the first schedule bandwidth change. I spent hours trying to understand what was happening, selectively removing parts of the code, adding debugging, etc. Wait()ing while locked was the only way I found to fix the problem (despite the fact that Wait should be thread safe). I also dislike the idea of locking tokenBucket for the entire duration, but that should not be a big issue, since the ticker is the only other main writer.